python/triton/language/semantic.py

```diff
@@ -1312,8 +1312,6 @@ def assert_dtypes_valid(lhs_dtype, rhs_dtype, options):
                 return
             assert lhs_dtype == rhs_dtype, f"First input ({lhs_dtype}) and second input ({rhs_dtype}) must have the same dtype!"
         else:
-            assert not lhs_dtype.is_fp8e4b15() and not rhs_dtype.is_fp8e4b15(
-            ), "Dot op does not support fp8e4b15 on CUDA arch >= 90"
```

Comment:
does that mean we don't check for this anymore?

Reply:
See Line 1332
```
    if lhs.dtype.is_fp8e4b15() or rhs.dtype.is_fp8e4b15():
        lhs = cast(lhs, tl.float16, builder)
        rhs = cast(rhs, tl.float16, builder)
```

I think we could cast it before checking types?

Reply:
right but on H100 I believe we wanted to explicit forbid it so that user don't generate inefficient code by mistake.

Reply:
I see. The problem is that `options.allow_fp8e4nv=True` doesn't always mean H100. The interpreter also has `options.allow_fp8e4nv=True`

Reply:
oh I see, yeah we should probably switch to a better check.